segmentation_labeling_app/transforms/transform_pipeline.py

```diff
@@ -0,0 +1,204 @@
+import argschema
+import marshmallow as mm
+from pathlib import Path
+import os
+import json
+import imageio
+import numpy as np
+import segmentation_labeling_app.utils.query_utils as query_utils
+from segmentation_labeling_app.rois.rois import ROI
+from segmentation_labeling_app.transforms.transformations import (
+        downsample_array, downsample_h5_video, transform_to_mp4)
+from segmentation_labeling_app.transforms.array_utils import (
+        content_extents)
+
+
+class TransformPipelineException(Exception):
+    pass
+
+
+class TransformPipelineSchema(argschema.ArgSchema):
+    segmentation_run_id = argschema.fields.Int(
+        required=True,
+        description=("which segmentation_run_id to transform. "
+                     "If not provided, will attempt to query using args: "
+                     "ophys_experiment_id, ophys_segmentation_commit_hash."))
+    ophys_experiment_id = argschema.fields.Int(
+        required=False,
+        description=("used as a query filter if "
+                     "segmentation_run_id is not provided."))
+    ophys_segmentation_commit_hash = argschema.fields.Str(
+        required=False,
+        description=("used as a query filter if "
+                     "segmentation_run_id is not provided."))
+    artifact_basedir = argschema.fields.OutputDir(
+        required=True,
+        description=("artifacts will be written to "
+                     "artifact_basedir/segmentation_run_id."))
+    cropped_shape = argschema.fields.List(
+        argschema.fields.Int,
+        required=True,
+        default=[128, 128],
+        description="[h, w] of bounding box around ROI images/videos")
+    quantile = argschema.fields.Float(
+        required=True,
+        default=0.1,
+        description=("quantile threshold for outlining an ROI. "))
+    input_fps = argschema.fields.Int(
+        required=False,
+        default=31,
+        description="frames per second of input movie")
+    output_fps = argschema.fields.Int(
+        required=False,
+        default=4,
+        description="frames per second of downsampled movie")
+    downsampling_strategy = argschema.fields.Str(
+        required=False,
+        default="average",
+        validator=mm.validate.OneOf(['average', 'first', 'last', 'random']),
+        description="what downsampling strategy to apply to movie and trace")
+    random_seed = argschema.fields.Int(
+        required=False,
+        default=0,
+        description="random seed to use if downsampling strategy is 'random'")
+
+    @mm.pre_load
+    def set_segmentation_run_id(self, data, **kwargs):
+        """if segmentation_run_id not provided, query the database
+        using the other provided args
+        """
+        if "segmentation_run_id" not in data:
```

Comment:
Is this alternative method easier, or is there another reason to include it? Seems like you'd have to query the segmentation_run_id table anyway to check the commit hash, unless you instead take the most recent run.

Reply:
I was just trying to anticipate a different use case. We can take it out to keep it simple.

Reply:
after discussion: we'll leave this in for now. Debatable how useful it is.

Reply:
It's just a nit. Fine to leave it in.